src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -232,4 +272,46 @@ private function formatWithIdentifiers(ObjectManager $em, ClassMetadata $class,
 
         return sprintf('object("%s") identified by (%s)', $idClass, implode(', ', $identifiers));
     }
+
+    public function getFieldValues($object, ClassMetadata $class, array $fields, bool $isEntity = false): array
+    {
+        if (!$isEntity) {
+            $reflectionObject = new \ReflectionObject($object);
+        }
+
+        $fieldValues = [];
+        $objectClass = \get_class($object);
+
+        foreach ($fields as $objectFieldName => $entityFieldName) {
+            if (!$class->hasField($entityFieldName) && !$class->hasAssociation($entityFieldName)) {
+                throw new ConstraintDefinitionException(sprintf('The field "%s" is not mapped by Doctrine, so it cannot be validated for uniqueness.', $entityFieldName));
+            }
+
+            $fieldName = \is_int($objectFieldName) ? $entityFieldName : $objectFieldName;
+            if (!$isEntity) {
+                if (!$reflectionObject->hasProperty($fieldName)) {
+                    throw new ConstraintDefinitionException(sprintf('The field "%s" is not a property of class "%s".', $fieldName, $objectClass));
+                }
+            }
+
+            $fieldValues[$entityFieldName] = $this->getPropertyValue($objectClass, $fieldName, $object);
+        }
+
+        return $fieldValues;
+    }
+
+    public function getPropertyValue(string $class, string $name, $object): mixed
+    {
+        $property = new \ReflectionProperty($class, $name);
+
+        if (!$property->isPublic()) {
+            $property->setAccessible(true);
```

Comment:
This is not required anymore since PHP 8.1.

Reply:
Removed